GameServerLib/GameObjects/GameObject.cs

```diff
@@ -80,25 +99,136 @@ public GameObject(Game game, float x, float y, int collisionRadius, int visionRa
             _toRemove = false;
         }
 
+        /// <summary>
+        /// Called by ObjectManager after AddObject (usually right after instatiation of GameObject).
+        /// </summary>
         public virtual void OnAdded()
         {
             _game.Map.CollisionHandler.AddObject(this);
         }
 
+        /// <summary>
+        /// Called by ObjectManager every tick.
+        /// </summary>
+        /// <param name="diff">Number of milliseconds that passed before this tick occurred.</param>
+        public virtual void Update(float diff)
+        {
+            if (Target != null && !Target.IsSimpleTarget)
+            {
+                Move(diff);
+            }
+            else
+            {
+                Move(diff);
+            }
+        }
+
+        /// <summary>
+        /// Whether or not the object should be removed from the game (usually both server and client-side). Refer to ObjectManager.
+        /// </summary>
+        public bool IsToRemove()
+        {
+            return _toRemove;
+        }
+
+        /// <summary>
+        /// Will cause ObjectManager to remove the object (usually) both server and client-side next update.
+        /// </summary>
+        public virtual void SetToRemove()
+        {
+            _toRemove = true;
+        }
+
+        /// <summary>
+        /// Called by ObjectManager after the object has been SetToRemove.
+        /// </summary>
         public virtual void OnRemoved()
         {
             _game.Map.CollisionHandler.RemoveObject(this);
         }
 
+        /// <summary>
+        /// Sets the server-sided position of this object.
+        /// </summary>
+        public virtual void SetPosition(float x, float y)
+        {
+            X = x;
+            Y = y;
+
+            Target = null;
+        }
+
+        /// <summary>
+        /// Sets the server-sided position of this object.
+        /// </summary>
+        public virtual void SetPosition(Vector2 vec)
+        {
+            X = vec.X;
+            Y = vec.Y;
+            Target = null;
+        }
+
+        /// <summary>
+        /// Refers to the height that the object is at in 3D space. *NOTE*: Should be renamed.
+        /// </summary>
+        // TODO: Change this to property to GetY() as it does not actually represent the Z-axis in 3D space, it currently represents the Y-axis
```

Comment:
Probably better stated as `GetHeight` as X,Y,Z are really depends on the orientation.